### PR TITLE
Pull Request

### PR DESCRIPTION
**Pull request by Textarea-LotC**
I didn't do any testing, but it should work.

### CODE REVIEW

**Comment by Archemedes** [comment]
this won't solve the color loss on line break.

**Comment by Textarea-LotC** [comment]
i'm a failure ¯\_(ツ)_/¯

**Review by Tofuus — CHANGES_REQUESTED** [review]
Check comments

**Comment by Archemedes** [comment]
Can you in countdown add an -r flag to alter BROADCAST_RADIUS? said flag should have a permission like "lever.countdown.radius". Remember to add this node to plugin.yml

**Comment by Archemedes** [comment]
Also if you're up to it you can try to investigate why /spectate -t wont stick to the target (client-side issue) but this isn't required

File: src/main/java/co/lotc/lever/cmd/Countdown.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import org.bukkit.entity.Player;
+
+import co.lotc.core.bukkit.util.LocationUtil;
+import co.lotc.lever.BaseCommand;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Countdown extends BaseCommand {
+    private static final int BROADCAST_RADIUS = 20;
```
Comment (Tofuus) [inline src/main/java/co/lotc/lever/cmd/Countdown.java]: This should shouldn't be a hard coded variable. Allow the user to set radius in the command

File: src/main/java/co/lotc/lever/cmd/Countdown.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    public void invoke(Player source, @Arg("Timer") @Default("5") @Range(min=5,max=60) int timer) {
+        List<Player> PlayersNearby = new ArrayList<>();
+        Bukkit.getOnlinePlayers().stream()
+                .filter(player->LocationUtil.isClose(source, player, BROADCAST_RADIUS))
+                .forEach(PlayersNearby::add);
+
+        Run.as(plugin).sync(()->PlayersNearby.forEach(player->player.sendMessage(ITALIC + "" + GOLD + source.getName() + DARK_GREEN + " has initiated a pvp timer, get ready!")));
```
Comment (Tofuus) [inline src/main/java/co/lotc/lever/cmd/Countdown.java]: Sending messages is safe to do async

File: src/main/java/co/lotc/lever/cmd/Countdown.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import java.util.List;
+
+public class Countdown extends BaseCommand {
+    private static final int BROADCAST_RADIUS = 20;
+
+    public void invoke(Player source, @Arg("Timer") @Default("5") @Range(min=5,max=60) int timer) {
+        List<Player> PlayersNearby = new ArrayList<>();
+        Bukkit.getOnlinePlayers().stream()
+                .filter(player->LocationUtil.isClose(source, player, BROADCAST_RADIUS))
+                .forEach(PlayersNearby::add);
```
Comment (Tofuus) [inline src/main/java/co/lotc/lever/cmd/Countdown.java]: I'm not 100% about this approach. This means players who move into range after the countdown started won't be receiving messages

File: src/main/java/co/lotc/lever/cmd/Emote.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    @Flag(name="r",description="Emote Radius",type=double.class)
+    @Flag(name="rp",description="Show RP name at the start of the emote")
+    public void invoke(Player source, @Arg("Emote Text") String emoteText) {
+        final double broadcastRadius = hasFlag("r") ? getFlag("r") : 64.0;
+
+        ChatBuilder b = new ChatBuilder(hasFlag("rp") ? source.getDisplayName() + " " : "[!] ").color(YELLOW).hover("Sent by " + source.getDisplayName() + ".")
+                .append(emoteText).color(YELLOW);
+        Bukkit.getOnlinePlayers().stream()
+                .filter(player-> LocationUtil.isClose(source, player, broadcastRadius))
+                .forEach(player->b.send(player));
```
Comment (Tofuus) [inline src/main/java/co/lotc/lever/cmd/Emote.java]: Be consistent here about whether you have a space after the arrow or not. Check the rest of the project and stay consistent with it

File: src/main/java/co/lotc/lever/cmd/ShowItem.java [hunk 5]
```diff
@@ -18,9 +18,11 @@ public void invoke(Persona source, Player target) {
 		validate(LocationUtil.isClose(source.getPlayer(), target, 16), "Can only show items to nearby players!");
 		ChatBuilder b = new ChatBuilder(source.getName()).color(GOLD)
 				.append(" is showing you ").color(AQUA)
-				.append('[' + ItemUtil.getDisplayName(is) + ']').color(WHITE).hoverItem(is);
-		
-		b.send(target);
+				.append('[').color(WHITE)
+				.append(ItemUtil.getDisplayName(is)).hoverItem(is)
```
Comment (Tofuus) [inline src/main/java/co/lotc/lever/cmd/ShowItem.java:22]: Still need to convert legacy text to chat components here

**Review by Tofuus — CHANGES_REQUESTED** [review]

**Comment by Tofuus** [comment]
Let's squash this down before we merge it in? You'll really pollute the git history if you merge as is

**Comment by Textarea-LotC** [comment]
New pull request for these commits: https://github.com/Archemedes/lEver/pull/2

File: src/main/java/co/lotc/lever/cmd/ShowItem.java (hunk 5 above)
Comment (Textarea-LotC) [reply src/main/java/co/lotc/lever/cmd/ShowItem.java:22]: idk how to do that :)

File: src/main/java/co/lotc/lever/cmd/Countdown.java (hunk 3 above)
Comment (Textarea-LotC) [reply src/main/java/co/lotc/lever/cmd/Countdown.java]: same, but i figured people would already know if someone was going to start a countdown :p

**Pull request closed without merging**